src/vcf/record.cpp

```diff
@@ -281,79 +320,97 @@ namespace ebi
                 format_meta.push_back(MetaEntry{line, ""});
             }
         }
+
+        return format_meta;
+    }
+
+    void Record::check_sample(size_t i, std::vector<MetaEntry> format_meta) const
+    {
+        std::vector<std::string> subfields;
+        util::string_split(samples[i], ":", subfields);
         
-        // Check the samples contents and accordance to the meta section
-        for (size_t i = 0; i < samples.size(); ++i) {
-            std::vector<std::string> subfields;
-            util::string_split(samples[i], ":", subfields);
-            
-            // The number of subfields can't be greater than the number in the FORMAT column
-            if (subfields.size() > format.size()) {
-                throw new SamplesBodyError{line, "Sample #" + std::to_string(i+1) +
-                        " has more fields than specified in the FORMAT column"};
-            }
-            
-            std::vector<std::string> alleles;
-            // If the first format field is not a GT, then no alleles need to be checked
-            if (format[0] == "GT") {
-                util::string_split(subfields[0], "|/", alleles);
+        check_sample_subfields_count(i, subfields);
+        
+        std::vector<std::string> alleles;
+        // If the first format field is not a GT, then no alleles need to be checked
+        if (format[0] == "GT") {
+            util::string_split(subfields[0], "|/", alleles);
+                
+            // The allele indexes must not be greater than the total number of alleles
+            check_samples_alleles(alleles);
+        }
```

Comment:
@cyenyxe, I'm thinking if it's worth having this `if` inside its own method, so that `check_sample` reads naturally that it does 3 types of checks:
- check_sample_subfields_count
- check_sample_gt_if_present
- check_sample_subfields_cardinality_type